contrib/init/upstart/docker.conf

```diff
@@ -70,3 +70,25 @@ post-start script
 		echo "$DOCKER_SOCKET is up"
 	fi
 end script
+
+post-stop script
+        SHUTDOWN_RUNLEVELS='06'
+        if runlevel | cut -d' ' -f2 | grep -q [$SHUTDOWN_RUNLEVELS]; then
+                for container in $(docker-runc list -q); do
+                        docker-runc kill "$container" || true
```

Comment:
@tianon I agree, it would be great to have docker itself handle this case, but until that happens, this is my workout with systems that use upstart.

I assume other init systems have the same problem, but haven't confirmed. My infrastructure currently only has systems that use upstart.